src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-directories/cephfs-directories.component.ts

```diff
@@ -227,29 +284,53 @@ export class CephfsDirectoriesComponent implements OnInit, OnChanges {
   }
 
   private setSettings(node: Tree) {
-    const files = this.getQuota(node, 'max_files');
-    const size = this.getQuota(node, 'max_bytes');
+    const readable = (value: number, fn?: (number) => number | string): number | string =>
+      value ? (fn ? fn(value) : value) : '';
+
     this.settings = [
-      {
-        name: 'Max files',
-        value: files.value,
-        origin: files.origin
-      },
-      {
-        name: 'Max size',
-        value: size.value !== '' ? this.dimlessBinaryPipe.transform(size.value) : '',
-        origin: size.origin
-      }
+      this.getQuota(node, 'max_files', readable),
+      this.getQuota(node, 'max_bytes', (value) =>
+        readable(value, (v) => this.dimlessBinaryPipe.transform(v))
+      )
     ];
   }
 
-  private getQuota(tree: Tree, quotaSetting: string): { value: string; origin: string } {
-    tree = this.getOrigin(tree, quotaSetting);
+  private getQuota(
+    tree: Tree,
+    quotaKey: string,
+    valueConvertFn: (number) => number | string
+  ): QuotaSetting {
+    // Get current maximum
+    const currentPath = tree.id;
+    tree = this.getOrigin(tree, quotaKey);
     const dir = this.getDirectory(tree);
-    const value = dir.quotas[quotaSetting];
+    const value = dir.quotas[quotaKey];
+    // Get next tree maximum
+    // => The value that isn't changeable through a change of the current directories quota value
+    let nextMaxValue = value;
+    let nextMaxPath = dir.path;
+    if (tree.id === currentPath) {
+      if (tree.parent.value === '/') {
+        // The value will never inherit any other value, so it has no maximum.
+        nextMaxValue = 0;
+      } else {
+        const nextMaxDir = this.getDirectory(this.getOrigin(tree.parent, quotaKey));
+        nextMaxValue = nextMaxDir.quotas[quotaKey];
+        nextMaxPath = nextMaxDir.path;
+      }
+    }
     return {
-      value: value ? value : '',
-      origin: value ? dir.path : ''
+      row: {
+        name: quotaKey === 'max_bytes' ? this.i18n('Max size') : this.i18n('Max files'),
```

Comment:
Maybe something for a follow up PR, don't want to block this PR from getting merged now.
```suggestion
        name: quotaKey === 'max_bytes' ? this.i18n('Max. size') : this.i18n('Max. files'),
```

Reply:
I tested it with 'Maximum' instead but that breaks the label in the modal. Only 'Max' is also used in pool and RBD quota settings.

For now I would leave it the current way.